ubi_manifest/worker/tasks/depsolve.py

```diff
@@ -0,0 +1,119 @@
+import logging
+from typing import Dict, List
+
+from ubi_manifest.worker.tasks.celery import app
+from ubi_manifest.worker.tasks.depsolver.models import DepsolverItem, UbiUnit
+from ubi_manifest.worker.tasks.depsolver.rpm_depsolver import Depsolver
+from ubi_manifest.worker.tasks.depsolver.ubi_config import UbiConfigLoader
+from ubi_manifest.worker.tasks.depsolver.utils import (
+    make_pulp_client,
+    remap_keys,
+    split_filename,
+)
+
+_LOG = logging.getLogger(__name__)
+
+
+@app.task
+def depsolve_task(ubi_repo_ids: List[str]) -> Dict[str, List[UbiUnit]]:
```

Comment:
Are you going to use a non-None return type for real, or is it just for testing?  My recollection is that having a celery task return a value has some significant caveats, for example at https://docs.celeryproject.org/en/stable/reference/celery.result.html , this warning about potential for leaks:

> you must eventually call get() or forget() on EVERY AsyncResult instance returned after calling a task

I don't quite get how the depsolve_task is going to be "plugged in" to an API endpoint yet.

Reply:
I guess this is not the final solution. 
My plan is to save the result of the depsolving into redis. I thought that web app could handle that, but when I'm thinking about it more, it would be best if the task saved result to redis so we can ignore the return value.  
The API endpoint would call this task and return task_id, caller waits until it's finished, then the result is acquired by the same or different API, that will look up the result in redis. Hope this makes sense.